packages/pwa-kit-extension-sdk/src/react/classes/ApplicationExtension.ts

```diff
@@ -30,6 +33,19 @@ export type ReactApplicationExtensionConfig = ApplicationExtensionConfig
 export class ApplicationExtension<
     Config extends ReactApplicationExtensionConfig
 > extends ApplicationExtensionBase<Config> {
+    protected _cachedRoutes: RouteProps[] | null = null
+
+    constructor(config: Config) {
+        super(config)
+
+        if (!isServerSide()) {
+            // Deserialize the routes on the client to ensure the latest routes are loaded on the client
+            this._cachedRoutes = this.deserializeAsyncRoutes() || null
```

Comment:
Why do we need this conditional null value. Is this._cachedRoutes specifically expecting a null value and not a null-ish value like undefined?

Reply:
removed!